### PR TITLE
Improve migration performance

### PR DESCRIPTION
**Pull request by codingjoe**
* perform delete IO async
* create only newly added sizes
* delete only obsolete file sizes
* create sizes from an in-memory copy of the file

### CODE REVIEW

File: tests/test_models.py [hunk 1]
```diff
@@ -38,6 +41,15 @@ class TestSimplePicture:
         width=800,
     )
 
+    def test_hash(self):
+        assert hash(self.picture_with_ratio) != hash(self.picture_without_ratio)
+        assert hash(self.picture_with_ratio) == hash(self.picture_with_ratio)
+
+    def test_eq(self):
+        assert self.picture_with_ratio != self.picture_without_ratio
+        assert self.picture_with_ratio == self.picture_with_ratio
```
Comment (amureki) [inline tests/test_models.py:50]: Last asserts look weird tbh. 1 == 1 is not adding much value (except for looking consistent).

**Review by amureki — APPROVED** [review]
Code-wise it looks good to me, however I did not test it. Not sure I'd manage testing before going offline for some tim

File: tests/test_models.py (hunk 1 above)
Comment (codingjoe) [reply tests/test_models.py:50]: I mean I am testing the equality function. So self == self is not a bad edge case to check.
